src/provider/provider_tracking.c

```diff
@@ -131,7 +131,9 @@ static umf_result_t trackingFree(void *hProvider, void *ptr, size_t size) {
     if (ptr) {
         ret = umfMemoryTrackerRemove(p->hTracker, ptr, size);
         if (ret != UMF_RESULT_SUCCESS) {
-            return ret;
+            // DO NOT return an error here, because the tracking provider
```

Comment:
I looked into `umfMemoryTrackerRemove` implementation and it can fail only if the state of the memory tracker is invalid. So probably we should add assert here and just crash.

Reply:
See: https://github.com/oneapi-src/unified-memory-framework/pull/141#issuecomment-1890259875
"It looks like older version of jemalloc called `arena_extent_dalloc()` with an address that was not coming from any previous `arena_extend_alloc()` for some reason."

With assert here and https://github.com/oneapi-src/unified-memory-framework/pull/141 merged all jemalloc_pool test would crash here, so it is not a good idea. 
We could add assert inside a debug ifdef, as a debug feature only .

Reply:
If `arena_extent_dalloc()` is called with an address that was not coming from any previous `arena_extend_alloc()` can we fix the memory tracker map? My point is that we should keep data in the tracker map in a valid state otherwise, the tracker is in the invalid state. I asked clarify question in [#141 (comment)](https://github.com/oneapi-src/unified-memory-framework/pull/141#issuecomment-1893582613)

Reply:
jemalloc sometimes calls `arena_extend_dalloc()` for an address that does not come from any previous `arena_extend_alloc()` call. How can we fix that? I think the memory tracker map is OK and it does not need to be fixed. It is just an issue in jemalloc.

Reply:
my guess was that if jemalloc calls `arena_extend_dalloc()` and passes a sub-range of previously allocated and added to the tracker region then we can fix the tracker state by removing this sub-range from the tracker. But if the completely different region is deleted with `arena_extend_dalloc()` then I agree with you - nothing can be done on our side.